internal/runners/reset/reset.go

```diff
@@ -126,16 +127,26 @@ func (r *Reset) Run(params *Params) error {
 		}
 	}
 
-	err = localcommit.Set(r.project.Dir(), commitID.String())
+	err = checkoutinfo.SetCommitID(r.project.Dir(), commitID.String())
 	if err != nil {
 		return errs.Wrap(err, "Unable to set local commit")
 	}
 
 	// Ensure the buildscript exists. Normally we should never do this, but reset is used for resetting from a corrupted
 	// state, so it is appropriate.
 	if r.cfg.GetBool(constants.OptinBuildscriptsConfig) {
-		if err := buildscript_runbit.Initialize(r.project.Dir(), r.auth); err != nil {
-			return errs.Wrap(err, "Unable to initialize buildscript")
+		err := buildscript_runbit.Initialize(r.project.Dir(), r.project.Owner(), r.project.Name(), r.auth)
+		if err != nil {
+			if errors.Is(err, buildscript.ErrOutdatedAtTime) {
+				// Remove the outdated build script and try again.
+				if err := buildscript_runbit.Remove(r.project.Dir()); err != nil {
+					return errs.Wrap(err, "Unable to remove outdated build script")
+				}
+				err = buildscript_runbit.Initialize(r.project.Dir(), r.project.Owner(), r.project.Name(), r.auth)
+			}
+			if err != nil {
+				return errs.Wrap(err, "Unable to initialize buildscript")
+			}
```

Comment:
This feels more complex than it needs to be. This is a reset, so we should just always remove the file if it exists and then initialize. Initialize should not have a care in the world about the buildscript already existing beyond obviously erroring out if it exists. Point being it's "Initialize" not "InitializeOrUpdate".